tmt/base.py

```diff
@@ -286,35 +286,46 @@ def overview(tree):
     @staticmethod
     def create(name, template, tree, force=False):
         """ Create a new test """
-        if Test._opt('dry'):
-            pass
-        # Create directory
+        # Create paths
+        # Create directory path
         if name == '.':
             directory_path = os.getcwd()
         else:
             directory_path = os.path.join(tree.root, name.lstrip('/'))
-            tmt.utils.create_directory(directory_path, 'test directory')
 
-        # Create metadata
+        # Create metadata path
         metadata_path = os.path.join(directory_path, 'main.fmf')
-        try:
-            tmt.utils.create_file(
-                path=metadata_path,
-                content=tmt.templates.TEST_METADATA[template],
-                name='test metadata',
-                force=force)
-        except KeyError:
-            raise tmt.utils.GeneralError(f"Invalid template '{template}'.")
 
-        # Create script
+        # Create script path
         script_path = os.path.join(directory_path, 'test.sh')
-        try:
-            content = tmt.templates.TEST[template]
-        except KeyError:
-            raise tmt.utils.GeneralError(f"Invalid template '{template}'.")
-        tmt.utils.create_file(
-            path=script_path, content=content,
-            name='test script', force=force, mode=0o755)
+
+        if Test._opt('dry'):
+            print(f'Directory {directory_path} was not created.')
+            print(f'Test metadata {metadata_path} was not created.')
+            print(f'Test script {script_path} was not created.')
+            print('No changes in workdir were executed.')
+        else:
+            # Create directory
+            tmt.utils.create_directory(directory_path, 'test directory')
```

Comment:
Instead of handling the `dry` mode three times separately in `Story`, `Plan` and `Test` it would be better to add support for dry mode to `tmt.utils.create_directory()` and `tmt.utils.create_file()` and then call them from here like this:

    tmt.utils.create_directory(directory_path, 'test directory', dry=Test._opt('dry))

This would save us a lot of unnecessary code duplication.